### PR TITLE
[css-ui-4] Add 'field-sizing' property

### PR DESCRIPTION
**Pull request by tkent-google**
See https://github.com/w3c/csswg-drafts/issues/7542

### CODE REVIEW

**Comment by tkent-google** [comment]
My plan is that more detailed behavior is described in the HTML standard.

**Comment by tabatkins** [comment]
Per <https://github.com/w3c/csswg-drafts/issues/7542#issuecomment-1685352461>, the names "normal" and "auto" aren't very illuminating here. (And, notably, as defined here they appear to be the exact opposite of what I expected them to be - I expected "normal" to be the legacy half-replaced behavior, while "auto" was the new 'act like other elements' behavior. That's a good argument for the names being bad!)

Maybe we should go with `legacy | auto` then? `auto` still isn't any clearer, but `legacy` clearly points to "how form elements *used to* size", so `auto` must be the new behavior. `legacy | normal` would also be fine.

**Comment by tkent-google** [comment]
@tabatkins Let's discuss keywords in https://github.com/w3c/csswg-drafts/issues/7542

**Comment by tkent-google** [comment]
Updated this PR for `fixed | content`.
https://github.com/w3c/csswg-drafts/issues/7542#issuecomment-1747803316

@frivoal would you take a loot at this please?

**Review by frivoal — APPROVED** [review]

File: css-ui-4/Overview.bs [hunk 1]
```diff
@@ -2452,6 +2470,60 @@ Effects of 'appearance' on Semantic Aspects of Elements</h4>
...
+	Percentages: N/A
+	Computed Value: as specified
+	Canonical order: per grammar
+	Animation type: discrete
+	</pre>
+
+	For the purpose of this specification,
+	an <dfn export>element with default preferred size</dfn> is an element
+	of which size is fixed regardless of its content size though its
+	[=preferred size property=] is 'auto'.
```
Comment (frivoal) [inline css-ui-4/Overview.bs:2491]:
```suggestion
	whose [=intrinsic size=] is fixed regardless of the size of its content.
```

File: css-ui-4/Overview.bs [hunk 2]
```diff
@@ -2452,6 +2470,60 @@ Effects of 'appearance' on Semantic Aspects of Elements</h4>
...
+	The host language defines which elements are applicable to it.
+	For example, in HTML <{textarea}> is an [=element with default preferred size=].
+
+	<dl dfn-type=value dfn-for=field-sizing>
+		<dt><dfn>fixed</dfn>
+		<dd>
+			The UA must apply the default preferred size to
+			an element if it is an
+			[=element with default preferred size=] and its
+			[=preferred size property=] value is 'auto'.
```
Comment (frivoal) [inline css-ui-4/Overview.bs:2501]:
```suggestion
			For [=element with default preferred size=],
			the UA must apply set the [=intrinsic size=]
			to the default preferred size defined by the host language for that element.
```

File: css-ui-4/Overview.bs [hunk 3]
```diff
@@ -2452,6 +2470,60 @@ Effects of 'appearance' on Semantic Aspects of Elements</h4>
...
+			Otherwise UA must behave the same as ''field-sizing/content''.
+		<dt><dfn>content</dfn>
+		<dd>
+			The UA must not apply the default preferred size to an element.
```
Comment (frivoal) [inline css-ui-4/Overview.bs:2505]:
```suggestion
			The UA must determine the element's [=intrinsic size=] based on its content,
			and must ignore any default preferred size defined by the host language for that element.
```

**Review by frivoal — CHANGES_REQUESTED** [review]
Unless I am missing something, it seems to me that we'd be better off expressing the difference between the two modes in terms of intrinsic size rather than talking directly about the preferred size properties. That gets the desired effect into all aspects of sizing, which is my understanding of the groups resolution.

I've made suggested edits implementing that change. Let me know if you think these work, or if I missed subtlety you were deliberately introducing by using the proposed wording.

**Comment by tkent-google** [comment]
> I've made suggested edits implementing that change. 

Thanks! I accepted all your suggestions.

**Comment by tkent-google** [comment]
@frivoal  would you take a loot at this again please?

**Review by frivoal — APPROVED** [review]

**Pull request merged**